rls/utils.py

```diff
@@ -24,3 +44,64 @@ def generate_rls_policy(cmd, definition, policy_name, table_name, expr) -> TextC
 
     else:
         raise ValueError(f'Unknown policy command"{cmd}"')
+
+
+def policy_changed_checker(db_policy: Policy, metadata_policy: Policy) -> bool:
+    temp_metadata_policy = metadata_policy.model_copy()
+    temp_metadata_policy.expression = add_bypass_rls_to_expr(metadata_policy.expression)
+
+    if isinstance(temp_metadata_policy.cmd, list):
+        temp_metadata_policy.cmd = Command(temp_metadata_policy.cmd[0])
+
+    return bool(db_policy == temp_metadata_policy)
+
+
+def normalize_sql_policy_expression(expression: str) -> str:
+    """
+    Normalizes a SQL expression for comparison by:
+    - Lowercasing all keywords.
+    - Removing unnecessary whitespace.
+    - Standardizing CAST syntax and other quirks.
+    """
+
+    # do the same thing sqlparse did but without sql parse
+    parsed: str = expression.lower()
+
+    # Remove any :: type casts with the type after it ( any word) \w+ like this
+    parsed = re.sub(r"::\w+", "", parsed)
+
+    # Remove as anyword from expression
+    parsed = re.sub(r"as \w+", "", parsed)
+
+    parsed = parsed.replace(" ", "")
+    parsed = parsed.replace("(", "")
+    parsed = parsed.replace(")", "")
+    # Replace "CAST(... AS TYPE)" with "::type" for uniformity
+    parsed = parsed.replace("cast", "")
+
+    return parsed
+
+
+def compare_between_policy_sql_expressions(
+    first_expression: str, second_expression: str
+) -> bool:
+    """
+    Compare two SQL expressions for equivalence by normalizing them.
+
+    Args:
+        expr1 (str): The first SQL expression.
+        expr2 (str): The second SQL expression.
+
+    Returns:
+        bool: True if the expressions are equivalent, False otherwise.
+    """
+    try:
+        # Normalize both expressions
+        normalized_expr1 = normalize_sql_policy_expression(first_expression)
```

Comment:
why do we need the try/except? I don't see any places where these could fail?

Reply:
oh it was for a different implementation of the function and forgot to remove it, will remove it now